kernel-rs/src/fs/lfs/inode.rs

```diff
@@ -247,13 +247,13 @@ impl InodeGuard<'_, Lfs> {
         }
 
         // 2. Write the imap to segment.
-        assert!(tx
-            .fs
-            .imap()
-            .set(self.inum, disk_block_no, &mut segment, ctx));
+        let mut imap = tx.fs.imap(ctx);
+        assert!(imap.set(self.inum, disk_block_no, &mut segment, ctx));
+        imap.free(ctx);
```

Comment:
I don't think it will cause any problems, but I changed it anyway just in case and to make things consistent with `Itable::alloc_inode`. Note that,
* if you were talking about deadlocks, I don't think it'll cause any problems since when we want to lock both `Segment` and `Imap`, we always do it in the order of `Segment` -> `Imap`.
* if its about race conditions, then I also think its fine since we already wrote to the imap and `segment.commit()` doesn't modify its content (just reads the buf and writes it to the disk). If we want to modify the imap's content, we need the segment guard anyway.

Reply:
If there is no problem, isn’t freeing the imap after committing the segment detrimental to the performance in a multi-threaded execution?

Reply:
Yes, that's right and actually, there's a few more places where I decided to be careful (even though it may be unnecessary) rather than aim for better performance. I think we should remove such waste of performance later after we confirmed we removed all deadlocks in lfs.